utils/http.go

```diff
@@ -28,74 +29,137 @@ type rootHandler struct {
 	context context.Context
 	trust   signed.CryptoService
 	//cachePool redis.Pool
+	imageNameAt ImageNameLocation
+}
+
+// ImageNameLocation is used to define where the auth code should read the image name
+// from. For most repo operations, it will be ImageInURL, i.e. it's the segments between
+// `/v2/` and `/_trust`. For the changefeed however, an image name is optional, and if
+// present will be found in the query string.
+type ImageNameLocation int
+
+// Constants for different image locations
+const (
+	ImageInURL = iota
+	ImageInQueryString
+	NoImageName
+)
+
+func parseImageName(loc ImageNameLocation, r *http.Request) (string, error) {
+	switch loc {
+	case ImageInURL:
+		vars := mux.Vars(r)
+		return vars["imageName"], nil
+	case ImageInQueryString:
+		qs := r.URL.Query()
+		imageName := qs.Get("filter")
+		if imageName == "" {
+			// no image name means global feed
+			imageName = "*"
+		}
+		return imageName, nil
+	case NoImageName:
```

Comment:
nit: I'm wondering if this should be renamed to something else, like `NoImageUsed`?  I'm proposing this because of the `if imageName == ""` in the case above